### PR TITLE
 Doc: TGZ + Multiversion

### LINKED ISSUE #192: [2pt] TGZ deploy doc

### PR DESCRIPTION
**Pull request by dokshina**
Closes #192

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -151,35 +151,74 @@ to learn how to set up topology using this role.
...
 Role variables are used to configure started instances, cluster topology,
 vshard bootstrapping, and failover.
 
-Configuration format is described in detail in the
-[configuration format](#configuration-format) section.
+Common variables:
 
-* `cartridge_package_path` (`string`, optional): path to Cartridge RPM package
-  (application name will be detected as package name);
 * `cartridge_app_name` (`string`): application name, required;
```
Comment (hackallcode) [inline README.md]:
```suggestion
* `cartridge_app_name` (`string`, required): application name;
```

File: README.md [hunk 2]
```diff
@@ -151,35 +151,74 @@ to learn how to set up topology using this role.
...
 * `cartridge_scenario` (`list-of-strings`, optional): list of steps to be launched
   (see [change scenario](#using-scenario) for more details)
 * `cartridge_custom_steps_dir` (`string`, optional, default: `null`): path to directory
   containing YAML files of custom steps (see [change scenario](#using-scenario) for more details)
 * `cartridge_custom_steps` (`list-of-dicts`, optional, default: `[]`): list of custom steps
   (see [change scenario](#using-scenario) for more details)
+
+Application package configuration:
+
+* `cartridge_package_path` (`string`, optional): path to Cartridge RPM package
```
Comment (hackallcode) [inline README.md]:
```suggestion
* `cartridge_package_path` (`string`, optional, default: `null`): path to Cartridge RPM package
```

File: README.md [hunk 3]
```diff
@@ -151,35 +151,74 @@ to learn how to set up topology using this role.
...
+  (application name will be detected as package name);
+* `cartridge_enable_tarantool_repo` (`boolean`, optional, default: `true`):
+  indicates if the Tarantool repository should be enabled (for packages with
+  open-source Tarantool dependency);
+
+[TGZ](./doc/tgz.md)-specific configuration:
+
+* `cartridge_multiversion` (`boolean`, optional, default: `false`): use [multiversion
+  approach](./multiversion.md) for TGZ package.
```
Comment (hackallcode) [inline README.md]:
```suggestion
  approach](/doc/multiversion.md) for TGZ package.
```

File: README.md [hunk 4]
```diff
@@ -151,35 +151,74 @@ to learn how to set up topology using this role.
...
+
+Application package configuration:
+
+* `cartridge_package_path` (`string`, optional): path to Cartridge RPM package
+  (application name will be detected as package name);
+* `cartridge_enable_tarantool_repo` (`boolean`, optional, default: `true`):
+  indicates if the Tarantool repository should be enabled (for packages with
+  open-source Tarantool dependency);
+
+[TGZ](./doc/tgz.md)-specific configuration:
```
Comment (hackallcode) [inline README.md]:
```suggestion
[TGZ](/doc/tgz.md)-specific configuration:
```

File: README.md [hunk 5]
```diff
@@ -151,35 +151,74 @@ to learn how to set up topology using this role.
...
+
+* `cartridge_multiversion` (`boolean`, optional, default: `false`): use [multiversion
+  approach](./multiversion.md) for TGZ package.
+
+* `cartridge_install_tarantool_for_tgz` (`boolean`, default: `false`): flag indicates
```
Comment (hackallcode) [inline README.md:188]: Here and in the same places:
```suggestion
* `cartridge_install_tarantool_for_tgz` (`boolean`, optional, default: `false`): flag indicates
```

File: doc/multiversion.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
+# Multiversion
+
+The default versioning approach for Tarantool Cartridge applications is quite simple:
+
+* only one version of application is installed on machine;
+* updating instance happens on instance restart.
+
+Such approach doesn't protect from accidental updating instance, that can lead to
+errors (for example, if router starts to use new schema earlier than storages).
+The solution is using multiversion approach (currently ot's fully supported only
```
Comment (hackallcode) [inline doc/multiversion.md]:
```suggestion
The solution is using multiversion approach (currently it's fully supported only
```

File: doc/multiversion.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+for [TGZ](./tgz.md) packages).
```
Comment (hackallcode) [inline doc/multiversion.md]:
```suggestion
for [TGZ](/doc/tgz.md) packages).
```

File: doc/multiversion.md [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+
+Multiversion approach:
+
+* several versions of application can be installed on machine;
+* each instance uses fixed version - it is achieved by using symbolic links;
+* updating instance is (see [`update_instance` step](./scenario.md#update_instance)):
```
Comment (hackallcode) [inline doc/multiversion.md]:
```suggestion
* updating instance (see [`update_instance` step API](./scenario.md#update_instance)) consists of:
```

File: doc/multiversion.md [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+  * moving it's link to a newest version of application;
+  * instance restart.
+
+## Configuration
+
+All versions of application are placed in `<cartridge_install_dir>/<app-name>-<version>`
+directory.
+
+Symbolic links for instances are placed in `cartridge_instances_dir/<app-name>.<instance-name>`
```
Comment (hackallcode) [inline doc/multiversion.md]:
```suggestion
Symbolic links for instances are placed in `<cartridge_instances_dir>/<app-name>.<instance-name>`
```

File: doc/scenario.md [hunk 10]
```diff
@@ -241,15 +241,17 @@ Output facts:
 
 ### update_instance
 
-Update instance links for a new version of package (if multiple package versions is enabled).
+Update instance links for a new version of package (if
+[multiversion approach](./multiversion.md) is enabled).
```
Comment (hackallcode) [inline doc/scenario.md]:
```suggestion
[multiversion approach](/doc/multiversion.md) is enabled).
```

File: doc/tgz.md [hunk 11]
```diff
@@ -0,0 +1,57 @@
+# Managing TGZ package
+
+Deploying TGZ archive is different from deploying RPM or DEB.
+It requires additional actions such as configuring systemd units.
+All these actions can be disabled.
+
+Using TGZ package allows to use [multiversion approach](./multiversion.md)
```
Comment (hackallcode) [inline doc/tgz.md]:
```suggestion
Using TGZ package allows to use [multiversion approach](/doc/multiversion.md)
```

File: doc/tgz.md [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+that can be helpful to perform rolling update correclty.
+This approach uses links to specify version of application that is used by each
+instance.
+
+## Unpacking TGZ
+
+TGZ package contains one directory named `<app-name>` that contains application.
```
Comment (hackallcode) [inline doc/tgz.md]: Let's use one style of variables names
```suggestion
TGZ package contains one directory named `cartridge_app_name` that contains application.
```

File: doc/tgz.md [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+### Application directories
+
+Application directories are created before unpacking.
+To configure them use
+
+* `cartridge_data_dir`
+* `cartridge_run_dir`
+* `cartridge_conf_dir`
+* `cartridge_install_dir`
+* `cartridge_instances_dir` in case of using [multiversion](./multiversion.md)
```
Comment (hackallcode) [inline doc/tgz.md]: What about to describe it?

File: README.md (hunk 2 above)
Comment (dokshina) [reply README.md]: I think that "optional" is enough.

File: README.md (hunk 4 above)
Comment (dokshina) [reply README.md]: Why?

File: README.md (hunk 5 above)
Comment (dokshina) [reply README.md:188]: I removed `optional` word specially. If variable has default value, it's obvious optional. I sure that this make doc more readable.

File: doc/multiversion.md (hunk 9 above)
Comment (dokshina) [reply doc/multiversion.md]: Changed to `{{ cartridge_instances_dir }}`.

File: doc/scenario.md (hunk 10 above)
Comment (dokshina) [reply doc/scenario.md]: It doesn't work.

File: doc/tgz.md (hunk 11 above)
Comment (dokshina) [reply doc/tgz.md]: The same as above - doesn't work.

File: doc/tgz.md (hunk 13 above)
Comment (dokshina) [reply doc/tgz.md]: I think that it's good to keep variables description in one place (in multiversion doc in this case) and use links in other places.

File: README.md (hunk 4 above)
Comment (hackallcode) [reply README.md]: It's a full path, it's works in any place in project

File: README.md (hunk 5 above)
Comment (hackallcode) [reply README.md:188]: Then let's remove `optional` word from other fileds with default

File: doc/multiversion.md (hunk 9 above)
Comment (hackallcode) [reply doc/multiversion.md]: Great!

File: doc/scenario.md (hunk 10 above)
Comment (hackallcode) [reply doc/scenario.md]: It works... For example, link to `config` description in https://github.com/tarantool/ansible-cartridge/blob/b6955eb64f616cdfa857a5e8525d44af76cbf077/doc/scenario.md#configure_instance

File: doc/tgz.md (hunk 13 above)
Comment (hackallcode) [reply doc/tgz.md]: What about to make link like `description of variables [here](/doc/multiversion.md)`?

File: README.md [hunk 14]
```diff
@@ -217,37 +217,37 @@ Application package configuration:
...
+* `cartridge_defaults` (`dict`, default: `{}`): default configuration
   parameters values for instances;
 * `config` (`dict`, required): [instance configuration](#instances);
-* `restarted` (`boolean`, optional): flag indicates if instance should be
+* `restarted` (`boolean`): flag indicates if instance should be
   restarted or not (if this flag isn't specified, instance will be restarted if
   it's needed to apply configuration changes);
-* `expelled` (`boolean`, optional, default: `false`): boolean flag that indicates if instance must be expelled from topology;
-* `stateboard` (`boolean`, optional, default: `false`): boolean flag that indicates
+* `expelled` (`boolean`, default: `false`): boolean f that indicates if instance must be expelled from topology;
```
Comment (hackallcode) [inline README.md]:
```suggestion
* `expelled` (`boolean`, default: `false`): boolean flag that indicates if instance must be expelled from topology;
```

File: README.md [hunk 15]
```diff
@@ -146,56 +146,108 @@ all:
...
+* `stateboard` (`boolean`, default: `false`): boolean flag that indicates
    that the instance is a [stateboard](#stateboard-instance);
-* `instance_start_timeout` (`number`, optional, default: 60): time in seconds to wait for instance to be started;
-* `instance_discover_buckets_timeout` (`number`, optional, default: 60): time in seconds
+* `instance_start_timeout` (`number`, default: 60): time in seconds to wait for instance to be started;
```
Comment (hackallcode) [inline README.md]:
```suggestion
* `instance_start_timeout` (`number`, default: `60`): time in seconds to wait for instance to be started;
```

File: README.md [hunk 16]
```diff
@@ -146,56 +146,108 @@ all:
...
+* `cartridge_wait_buckets_discovery` (`boolean`, default: `true`): boolean
+  flag that indicates if routers should wait for buckets discovery after vshard bootstrap;
+* `instance_discover_buckets_timeout` (`number`, default: 60): time in seconds
```
Comment (hackallcode) [inline README.md]:
```suggestion
* `instance_discover_buckets_timeout` (`number`, default: `60`): time in seconds
```

**Review by hackallcode — APPROVED** [review]
Good documentation!
Only some small mistakes should be corrected.

**Pull request merged**